src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs

```diff
@@ -55,18 +56,36 @@ public SymbolTableOverRecordType(RecordType type, ReadOnlySymbolTable parent = n
         IEnumerable<KeyValuePair<string, NameLookupInfo>> IGlobalSymbolNameResolver.GlobalSymbols
         {
             get
-            {
-                foreach (var kv in _type.GetFieldTypes())
-                {
-                    yield return new KeyValuePair<string, NameLookupInfo>(kv.Name, Create(kv.Name, kv.Type));
+            {
+                if (_type._type.AssociatedDataSources.Any())

```

Comment:
Can you add a comment about what this is doing?
It's ok to also link to the bug 1212 in the comments here for future followup.